projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-component.scss

```diff
@@ -141,6 +145,69 @@
         @extend %grid__cbx-selection !optional;
     }
 
+    // Groupby section
+
+    @include e(group-label) {
+        @extend %grid__group-label !optional;
+    }
+
+    @include e(group-text) {
+        @extend %grid__group-text !optional;
+    }
+
+    @include e(group-count) {
+        @extend %grid__group-count !optional;
+    }
+
+    @include e(group-content) {
+        @extend %grid__group-content !optional;
+    }
+
+    @include e(group-indentation) {
+        @extend %grid__group-indentation !optional;
+    }
+
+    @include e(grouparea) {
+        @extend %grid-grouparea !optional;
+    }
+
+    @include e(group-field) {
+        @extend %grid__group-field !optional;
+    }
+
+    @include e(group-count) {
+        @extend %grid__group-count !optional;
+    }
+
+    @include e(grouping-indicator) {
+        @extend %igx-grid__grouping-indicator !optional;
+    }
+
+    @include e(group-indentation-header) {
+        @extend %igx-grid__group-indentation-header !optional;
+    }
+
+    @include e(group-expand-btn) {
+        @extend %igx-grid__group-expand-btn !optional;
+    }
+
+    @include e(group-icon) {
+        @extend %igx-grid__group-icon !optional;
+    }
+
+    @include e(grouparea-placeholder) {
+        @extend %igx-grid__grouparea-placeholder !optional;
+    }
+
+    @for $i from 1 through 10 {
+        @include e(group-indentation-level-#{$i}) {
+            @extend %grid__group-indentation-level-#{$i} !optional;
+        }
+        @include e(group-indentation-padding-level-#{$i}) {
+            @extend %grid__group-indentation-padding-level-#{$i} !optional;
+        }
+    }
+
```

Comment:
Lines 148 - 200 break the BEM notation standard.

Reply:
I don't see how this breaks BEM, please suggest a change.

Reply:
Well, each of those declarations signify direct relationship with the igx-grid element, which is not true. All of them state that they are direct ascendents of the igx-grid element. Furthermore, as written, they show no relationship between one-another, which means that removing any of the declaration will not affect the behavior or presentation of any other of the declared units, which is also not true. For instance, removing the group-label declaration will change both the behavior and the presentation of the group-field. I could find numerous similar relationships, which are not clear from how the code is written, which leads to confusion and renders using BEM useless.

Reply:
From what you say here It turns out that the hole groupby section should be separate (b)block not (e)element of the grid, am I correct?
I personally don't feel any difficulty to understand what is what here. For me, grouping is part of the grid and you cannot use it outside of the grid component. Another thing to keep in mind is that for example, the background color for the grouping indicators and the grouping row should match the background of the headers which means that it should have access to the same maps, variables...